examples/dspy-demo/utils.py

```diff
@@ -0,0 +1,124 @@
+from dsp import dotdict
+from typing import Union, List, Optional
+import dspy
+from sentence_transformers import SentenceTransformer
+from tidb_vector.integrations import TiDBVectorClient
+
+
+def sentence_transformer_embedding_function(
+        embed_model: SentenceTransformer,
+        sentences: Union[str, List[str]]
+) -> List[float]:
```

Comment:
return `List[List[float]]`?

Reply:
I need to first verify that model embeding handles two different signature types: List[str] and List[List[str]]. 
In addition, tidb_vector_client.query supports signature vector but not vectors.

I need to consider how to deal with it.